### PR TITLE
guard against issues where we try to access tensor_meta on non-tensor input

### PR DESCRIPTION
**Pull request by Chillee**
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* __->__ #82611

### CODE REVIEW

File: functorch/functorch/_src/partitioners.py [hunk 1]
```diff
@@ -333,7 +338,7 @@ def ban_recomputation(node):
             # then we don't allow recomputation.
             if 'tensor_meta' not in node.meta:
                 return False
-            input_tensors_size = sum(_size_of(i.meta['tensor_meta']) for i in node.args if isinstance(i, fx.Node))
+            input_tensors_size = sum(_maybe_size_of(i) for i in node.args if isinstance(i, fx.Node))
             output_size = _size_of(node.meta['tensor_meta'])
```
Comment (voznesenskym) [inline functorch/functorch/_src/partitioners.py:342]: does this node.meta always have a `tensor_meta` - or does it need the same treatment as above, `_maybe_size_of`?

**Review by voznesenskym — APPROVED** [review]

**Comment by Chillee** [comment]
Was failing on master.

**Pull request closed without merging**